op-challenger/game/fault/preimages/split.go

```diff
@@ -8,31 +8,26 @@ import (
 
 var _ PreimageUploader = (*SplitPreimageUploader)(nil)
 
-// PREIMAGE_SIZE_THRESHOLD is the size threshold for determining whether a preimage
-// should be uploaded directly or through the large preimage uploader.
-// TODO(client-pod#467): determine the correct size threshold to toggle between
-//
-//	the direct and large preimage uploaders.
-const PREIMAGE_SIZE_THRESHOLD = 136 * 128
-
 // SplitPreimageUploader routes preimage uploads to the appropriate uploader
 // based on the size of the preimage.
 type SplitPreimageUploader struct {
-	directUploader PreimageUploader
-	largeUploader  PreimageUploader
+	largePreimageSizeThreshold uint64
+	directUploader             PreimageUploader
+	largeUploader              PreimageUploader
 }
 
-func NewSplitPreimageUploader(directUploader PreimageUploader, largeUploader PreimageUploader) *SplitPreimageUploader {
-	return &SplitPreimageUploader{directUploader, largeUploader}
+func NewSplitPreimageUploader(directUploader PreimageUploader, largeUploader PreimageUploader, minLargePreimageSize uint64) *SplitPreimageUploader {
+	return &SplitPreimageUploader{minLargePreimageSize, directUploader, largeUploader}
 }
 
 func (s *SplitPreimageUploader) UploadPreimage(ctx context.Context, parent uint64, data *types.PreimageOracleData) error {
 	if data == nil {
 		return ErrNilPreimageData
 	}
-	if len(data.OracleData) > PREIMAGE_SIZE_THRESHOLD {
-		return s.largeUploader.UploadPreimage(ctx, parent, data)
-	} else {
+	// Always route local preimage uploads to the direct uploader.
+	if uint64(len(data.OracleData)) < s.largePreimageSizeThreshold || data.IsLocal {
```

Comment:
I'm fine with merging this as it does make sense to hard code local images as using direct upload. However, given `OracleData` should be `nil` for local data (since we only need to provide the key and the data from from the game contract), why would it ever use large preimages?